code/datums/langchat/langchat.dm

```diff
@@ -151,6 +151,43 @@
 	else
 		addtimer(CALLBACK(src, /atom.proc/langchat_drop_image, language), timer, TIMER_OVERRIDE|TIMER_UNIQUE|TIMER_NO_HASH_WAIT)
 
+//used for broadcasting from a vehicle
+/mob/proc/langchat_vehicle_broadcast(message, var/list/listeners, language, var/obj/vehicle/multitile/Vehicle)
+
+	if(!Vehicle || interactee != Vehicle)
+		return
+
+	langchat_drop_image()
+	langchat_make_image()
+
+	var/text_left = null
+	var/text_to_display = message
+
+	if(length(message) > LANGCHAT_LONGEST_TEXT)
+		text_to_display = copytext_char(message, 1, LANGCHAT_LONGEST_TEXT - 5) + "..."
+		text_left = "..." + copytext_char(message, LANGCHAT_LONGEST_TEXT - 5)
+	var/timer = 6 SECONDS
+	if(text_left)
+		timer = 4 SECONDS
+	text_to_display = "<span class='center [langchat_styles] langchat_announce langchat'>[text_to_display]</span>"
+
+	langchat_image.maptext = text_to_display
+	langchat_image.maptext_width = LANGCHAT_WIDTH * 2
+
+	langchat_listeners = listeners
+	for(var/mob/Listener in langchat_listeners)
+		if(langchat_client_enabled(Listener) && !Listener.ear_deaf && Listener.say_understands(src, language))
+			Listener.client.images += langchat_image
+
+	langchat_image.loc = interactee
+	langchat_image.maptext_y += 32
```

Comment:
this might need to be reset

Reply:
Resetting didn't work, had to figure out other way.

Reply:
that doesn't look correct to me because the maptext offset will revert after first message if it's a long message segmented

one solution i could see is to mark the langchat_image as "expendable" and if it is, delete it completely after display. that way a new one is created from scratch next time.